spec/support/server.rb

```diff
@@ -21,23 +21,24 @@ def to_h
   end
 end
 
-class DiagnoseServer < Sinatra::Base
+class MockServer < Sinatra::Base
   class << self
     def run!(port)
       @mutex = Mutex.new
-      @received_requests = []
+      @auth_response_code = 200
+      @received_diagnose_requests = []
       super(:port => port)
     end
 
-    def track_request(request)
+    def track_diagnose_request(request)
       @mutex.synchronize do
-        @received_requests << request
+        @received_diagnose_requests << request
       end
     end
 
-    def last_received_report
+    def last_received_diagnose_report
```

Comment:
The method name is quite long now, so consider dropping the `received`? (if so, then maybe also from the `@received_diagnose_requests` instance variable)